### PR TITLE
fix(waiting-executions) : concurrent waiting executions doesn't follow FIFO

### PR DESCRIPTION
**Pull request by rjalander**
Fix as discussed in the code PR https://github.com/spinnaker/orca/pull/4356#issuecomment-1480021591
for the new feature **Add support for max concurrent pipeline executions**

Due to the new feature **Add support for max concurrent pipeline executions** https://github.com/spinnaker/orca/pull/4193, now the waiting executions exists irrespective of concurrent execution is enabled/disabled 

So removed the condition to check for `isLimitConcurrent` is true/false as part of existing fix https://github.com/spinnaker/orca/pull/4356

### CODE REVIEW

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteExecutionHandler.kt [hunk 1]
```diff
@@ -85,17 +85,13 @@ class CompleteExecutionHandler(
...
-        } else {
-            log.debug("Execution ${execution.id} is not Disabled for concurrent executions, no need to run waiting executions")
+      if (execution.status != RUNNING) {
+        execution.pipelineConfigId?.let {
+          queue.push(StartWaitingExecutions(it, purgeQueue = !execution.isKeepWaitingPipelines))
         }
+      } else {
+        log.debug("Not starting waiting executions as execution ${execution.id} is currently RUNNING.")
+      }
       }
```
Comment (mattgogerly) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteExecutionHandler.kt:95]: Could you fix the indentation here please?

Comment (rjalander) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteExecutionHandler.kt:94]: Done

**Comment by sergio-quintero** [comment]
@mergifyio backport release-1.30.x

**Comment by link108** [comment]
@mergifyio backport release-1.30.x

**Pull request merged**